pkg/istio/onboarding/controller.go

```diff
@@ -187,26 +181,40 @@ func (c *Controller) sync() error {
 
 	if config == nil {
 		log.Println("Creating cluster rbac config...")
-		return c.createClusterRbacConfig(serviceList)
+		_, err := c.store.Create(createClusterRbacConfig(serviceList))
+		return err
 	}
 
 	clusterRbacConfig, ok := config.Spec.(*v1alpha1.RbacConfig)
 	if !ok {
 		return errors.New("Could not cast to ClusterRbacConfig")
 	}
 
+	needsUpdate := false
+	if clusterRbacConfig.Inclusion == nil || clusterRbacConfig.Mode != v1alpha1.RbacConfig_ON_WITH_INCLUSION {
+		log.Println("ClusterRBacConfig inclusion field is nil or ON_WITH_INCLUSION mode is not set, syncing...")
+		clusterRbacConfig = createClusterRbacSpec(serviceList)
+		needsUpdate = true
```

Comment:
in this case, since we already have the entire `serviceList`, we can directly call `c.store.Update()` without having to compare the diff right?

Reply:
Yes, I wanted to avoid two update calls in the function. I have updated it to directly call update and removed the needsUpdate flag. This logic will change anyway as we will be adding to the queue which does the api calls here instead.